src/util.rs

```diff
@@ -0,0 +1,24 @@
+use core::sync::atomic::{AtomicBool, Ordering};
+
+#[derive(Debug)]
+pub struct LazyBool {
+    value: AtomicBool,
+    needs_init: AtomicBool,
+}
+
+impl LazyBool {
+    pub const fn new() -> Self {
+        LazyBool {
+            value: AtomicBool::new(false),
+            needs_init: AtomicBool::new(true),
+        }
+    }
+
+    pub fn unsync_init(&self, init: impl FnOnce() -> bool) -> bool {
+        if self.needs_init.load(Ordering::SeqCst) {
+            self.value.store(init(), Ordering::SeqCst);
+            self.needs_init.store(false, Ordering::SeqCst);
```

Comment:
This isn't a "lazy bool" the way one may expect from `lazy_static`: it may run `init()` multiple times. Do we care about this, e.g. if someone spins up a lot of threads which each use `getrandom` simultaneously? See https://github.com/rust-random/rand/issues/804

Reply:
The description mentions this a little, but here we shouldn't care. The call to check `getrandom(2)` support is non-blocking and requests no randomness. So it would be about as quick as any other linux syscall that doesn't really do anything. And probably quicker than waiting on a kernel lock. 

Should the name be changed?

Reply:
I added some comments to better explain this.